src/configuration/app/size.md

```diff
@@ -1,27 +1,93 @@
+The plan size can be one of:
 
-Note that in a development environment this value is ignored and always set to `S`.  It will only take effect in a production deployment (a master branch with an associated domain).  If the total resources requested by all apps and services is larger than what the plan size allows then a production deployment will fail with an error.
+* Small
+* Standard
+* Medium
+* Large
+* X-Large
+* 2X-Large
 
-# How do I make a background processing container smaller to save resources?
+Each comes with an increasing CPU and memory pool that gets shared by all of the containers in your production environment.  That means that, in practice, the memory available on an individual application is far less than the Plan's total memory; that memory is divided up between the application, database, caching server, etc, and those backend services are often more memory-intensive than the application.
 
-Simply set the `size` key to `S` to ensure that the container gets fewer resources, leaving more to be allocated to other containers.
+Additionally, a "Development" plan has no production resourcing, and all environments are treated like development branches.
 
-```yaml
-name: processing
+## Service sizes
 
-type: nodejs:6.11
-size: S
+Services (those defined in `services.yaml`) also come in one of a number of sizes: `S`, `M`, `L`, `XL`, `2XL`, `4XL`.
 
-...
+These sizes have a fixed amount of CPU and RAM associated with them.
+
+Service sizes are set via the `size` property in `services.yaml`.  If not set, the default value is `AUTO`.
+
+## Application sizes
+
+Applications (those defined by the presence of a `.platform.app.yaml`) also come in one of a number of sizes: `XS`, `S`, `M`, `L`, `XL`, `2XL`, `4XL`.
+
+These sizes have a fixed amount of CPU associated with them, and a partially configurable amount of RAM.
+
+| Size | CPU  |
+|------|------|
+|  XS  | 0.25 |
+|  S   | 0.5  |
+|  M   | 1.0  |
+|  L   | 2.0  |
+|  XL  | 4.0  |
+|  2XL | 8.0  |
+
+The application size may be set via the `size` property in `.platform.app.yaml`.  If not set, the default value is `AUTO`.
+
+Note: The `XS` container size is only available if you specify a `resource` block as described below.
+
+## Application memory
+
+Application containers have memory allocated to them based on their container size as well, but the ratio is configurable.  Specifically, there are two properties that can be adjusted.  Their default values are shown below.
+
+```php
+resources:
+    base_memory: 64
+    memory_ratio: 256
 ```
+
+The `base_memory` is a fixed size (in megabytes) that the container will have, period.  The `memory_ratio` is a multiplier based on the CPU size.  That is, an `S` CPU size is 0.5 CPU slides, with the above configuration it would get 64 + 0.5 * 256 = 192 MB of RAM.  If the same application had its size increased to an XL, it would get 64 + 4 * 256 = 1088 MB of RAM.
```

Comment:
These `*` characters don't render for some reason. Maybe switch to parenthesis or `x`?

Reply:
Ugh.  It's probably trying to render them as formatting codes.  I hate Markdown.